pkg/operator/ceph/csi/spec.go

```diff
@@ -511,6 +579,15 @@ func createCSIDriverInfo(clientset kubernetes.Interface, name string, ownerRef *
 	return err
 }
 
+// deleteCSIDriverInfo deletes CSIDriverInfo and returns the error if any
+func deleteCSIDriverInfo(clientset kubernetes.Interface, name string) error {
+	err := clientset.StorageV1beta1().CSIDrivers().Delete(name, &metav1.DeleteOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
The logic is a bit backward, a more common/readable pattern is to check for `err != nil` then for `apierrors.IsNotFound(err)`. So if not `apierrors.IsNotFound(err)` return `err` instead of `nil` and exit with `nil` at the end.

Reply:
> The logic is a bit backward, a more common/readable pattern is to check for `err != nil` then for `apierrors.IsNotFound(err)`. So if not `apierrors.IsNotFound(err)` return `err` instead of `nil` and exit with `nil` at the end.
Okay, I'll first check for `err != nil` then for `apierrors.IsNotFound(err)` and  if ` if not apierrors.IsNotFound(err)` then we are returning `nil`  instead `err` because if there is no resource no reason to return err. We have discussed this [here](https://github.com/rook/rook/pull/5605#discussion_r445312414)